mslib/msui/mscolab.py

```diff
@@ -259,22 +279,38 @@ def text_changed(self):
         self.loginButton.setEnabled(self.emailid.text() != "" and self.password.text() != "")
 
     def handle_import(self):
-        file_path = get_open_filename(self, "Select a file", "", "Flight track (*.ftml)")
+        if self.workLocallyCheckBox.isChecked():
+            file_path = get_open_filename(self, "Select a file", "", "Flight track (*.ftml)")
+        else:
+            file_type = ["Flight track (*.ftml)"] + [f"Flight track (*.{ext})" for ext in self.import_plugins.keys()]
+            file_path = get_open_filename(self, "Select a file", "", ';;'.join(file_type))
         if file_path is None:
             return
         dir_path, file_name = fs.path.split(file_path)
-        with open_fs(dir_path) as file_dir:
-            xml_content = file_dir.readtext(file_name)
-        try:
-            model = ft.WaypointsTableModel(xml_content=xml_content)
-        except SyntaxError:
-            show_popup(self, "Import Failed", f"The file - {file_name}, does not contain valid XML")
-            return
-        self.waypoints_model = model
-        if self.workLocallyCheckBox.isChecked():
-            self.waypoints_model.save_to_ftml(self.local_ftml_file)
-            self.waypoints_model.dataChanged.connect(self.handle_waypoints_changed)
+        file_name = fs.path.basename(file_path)
```

Comment:
does this work on windows? fs.path seems to work only on unix/fs-style paths

Reply:
I assume it does, but will look up windows before the next  release ;)